integration tests for tracing (#174)
agera-edc/DataSpaceConnector: system-tests/README.md

# System tests

## Tracing integration tests

The [TracingIntegrationTests](./tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/FileTransferIntegrationTest.java) makes sure that tracing works correctly when triggering a file transfer.
To be able to run the `TracingIntegrationTests` locally, you need to place the [opentelemetry java agent](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v1.12.0/opentelemetry-javaagent.jar) in the project root folder.
Then you can run the tests:
```bash
./gradlew -p system-tests/tests test -DincludeTags="OpenTelemetryIntegrationTest"

> ```

Reviewer: It would be good to add details on how and why the `OtlpGrpcServer` is used for these tests.

# System tests

## Tracing integration tests

> The [TracingIntegrationTests](./tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/FileTransferIntegrationTest.java) makes sure that tracing works correctly when triggering a file transfer.

Reviewer 2: grammar singular or plural?

> This test triggers a file transfer with the opentelemetry java agent attached. The default trace exporter is based is the OTLP exporter based on gRPC protocol. That's why an OtlpGrpcServer is started to collect the traces.

Reviewer 2: ```
[opentelemetry java agent attached](link to doc page)
```

> This test triggers a file transfer with the [opentelemetry java agent attached](https://github.com/open-telemetry/opentelemetry-java-instrumentation). The default trace exporter is based is the OTLP exporter based on gRPC protocol. That's why an OtlpGrpcServer is started to collect the traces.

Reviewer 2: ```suggestion
This test triggers a file transfer with the [opentelemetry java agent attached](https://github.com/open-telemetry/opentelemetry-java-instrumentation). The default trace exporter configured in the agent is the OTLP exporter based on gRPC protocol. Therefore, a test OtlpGrpcServer is started to collect the traces.
```

To be able to run the `TracingIntegrationTests` locally, you need to place the [opentelemetry java agent](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v1.12.0/opentelemetry-javaagent.jar) in the project root folder.
Then you can run the tests:
```bash

> ./gradlew -p system-tests/tests test -DincludeTags="OpenTelemetryIntegrationTest"

Reviewer 2: ```suggestion
./gradlew :system-tests:tests:test -DincludeTags="OpenTelemetryIntegrationTest"
```

> ./gradlew :system-tests:tests:test -DincludeTags="OpenTelemetryIntegrationTest"

Reviewer 3: nit: with -p just for consistency.
